edb/schema/types.py

```diff
@@ -2544,6 +2944,53 @@ def get_dummy_expr_field_value(
             raise AssertionError(f'unhandled field {field.name!r}')
 
 
+class CreateRange(CreateCollectionType[Range]):
+    pass
+
+
+class AlterRange(AlterCollectionType[Range]):
+    pass
+
+
+class RenameRange(RenameCollectionType[Range]):
+    pass
+
+
+class CreateRangeExprAlias(CreateCollectionExprAlias[RangeExprAlias]):
+    def _get_ast_node(
+        self, schema: s_schema.Schema, context: sd.CommandContext
+    ) -> typing.Type[qlast.CreateAlias]:
+        # Can't just use class-level astnode because that creates a
+        # duplicate in ast -> command mapping.
+        return qlast.CreateAlias
+
+
+class RenameRangeExprAlias(
+    CollectionExprAliasCommand[RangeExprAlias],
+    sd.RenameObject[RangeExprAlias],
+):
+    pass
+
+
+class AlterRangeExprAlias(
+    CollectionExprAliasCommand[RangeExprAlias],
+    sd.AlterObject[RangeExprAlias],
+):
+
+    def get_dummy_expr_field_value(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+        field: so.Field[Any],
+        value: Any,
+    ) -> Optional[s_expr.Expression]:
+        # XXX: what's this dummy value?
```

Comment:
This is needed for some switcheroo shenanigans when we do alter propagation in expressions.  Using `range()` should be fine.